event_listener/dfusion_db/snapp_event_receiver.py

```diff
@@ -59,29 +65,51 @@ def update_accounts(self, event: Dict[str, Union[int, str, str, int]]):
         balances = self.db.get_account_state(state_index - 1)['balances']
         num_tokens = self.db.get_num_tokens()
 
-        if transition_type == 0:  # Deposit
-            applied_deposits = self.db.get_deposits(event['slot'])
-            for deposit in applied_deposits:
-                a_id = deposit['accountId']
-                t_id = deposit['tokenId']
-                amount = deposit['amount']
-                self.logger.info("Incrementing balance of account {} - token {} by {}".format(a_id, t_id, amount))
+        applied_data = self.get_data_to_apply(transition_type, event['slot'])
+
+        for datum in applied_data:
+            a_id = datum['accountId']
+            t_id = datum['tokenId']
+            amount = datum['amount']
+
+            # Balances are stored as [b(a1, t1), b(a1, t2), ... b(a1, T), b(a2, t1), ...]
+            index = num_tokens * (a_id - 1) + (t_id - 1)
+
+            if transition_type == self.TRANSITION_TYPES['Deposit']:
 
-                # Balances are stored as [b(a1, t1), b(a1, t2), ... b(a1, T), b(a2, t1), ...]
+                self.logger.info("Incrementing balance of account {} - token {} by {}".format(a_id, t_id, amount))
                 balances[num_tokens * (a_id - 1) + (t_id - 1)] += amount
```

Comment:
Could use the index declared above here for consistency. (I forgot when reorganizing)